app/modules/asset_groups/models.py

```diff
@@ -807,34 +866,32 @@ def complete(self):
         for job_id in self.jobs:
             assert not self.jobs[job_id]['active']
 
-        self.stage = AssetGroupSightingStage.processed
-        with db.session.begin(subtransactions=True):
-            db.session.merge(self)
-        db.session.refresh(self)
+        self.set_stage(AssetGroupSightingStage.processed)
 
     def job_complete(self, job_id_str):
         if job_id_str in self.jobs:
-            self.jobs[job_id_str]['active'] = False
-
-            outstanding_jobs = []
-            for job in self.jobs.keys():
-                if self.jobs[job]['active']:
-                    outstanding_jobs.append(job)
-
-            if len(outstanding_jobs) == 0:
-                # All complete, updatedAssets now in same sate as other assets so no need to store anymore
-                if 'updatedAssets' in self.config:
-                    del self.config['updatedAssets']
-                self.config = self.config
-                self.stage = AssetGroupSightingStage.curation
-                self.curation_start = datetime.utcnow()
-
-            # This is necessary because we can only mark jobs as
-            # modified if we assign to it
-            self.jobs = self.jobs
-
             with db.session.begin(subtransactions=True):
+                self.jobs[job_id_str]['active'] = False
```

Comment:
curious to learn why this needed to be in a db.session

Reply:
There are multiple attributes being changed here between `self.jobs` and `self.config`.  The transaction is not long, so I encapsulated the entire block because some state was not being recorded into the database (specifically active = False).